--- conf/tachyon-env.sh.template	2015-04-24 15:23:49.000000000 -0700
+++ /home/tianyin/tachyon-0.6.4-test/tachyon-0.6.4/conf/tachyon-env.sh	2015-05-31 19:42:10.215604895 -0700
@@ -46,11 +46,12 @@
   export JAVA_HOME="$(dirname $(which java))/.."
 fi
 
+export JAVA_HOME="/usr/lib/jvm/java-1.7.0-openjdk-amd64"
 export JAVA="$JAVA_HOME/bin/java"
 export TACHYON_MASTER_ADDRESS=localhost
 export TACHYON_UNDERFS_ADDRESS=$TACHYON_HOME/underfs
 #export TACHYON_UNDERFS_ADDRESS=hdfs://localhost:9000
-export TACHYON_WORKER_MEMORY_SIZE=1GB
+export TACHYON_WORKER_MEMORY_SIZE=10942842KB
 export TACHYON_UNDERFS_HDFS_IMPL=org.apache.hadoop.hdfs.DistributedFileSystem
 
 CONF_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
@@ -73,10 +74,28 @@
   -Dtachyon.master.journal.folder=$TACHYON_HOME/journal/
   -Dorg.apache.jasper.compiler.disablejsr199=true
   -Djava.net.preferIPv4Stack=true
+  -XX:+PrintGCDetails
+  -XX:+PrintGCDateStamps
 "
-
+#  -Dcom.sun.management.jmxremote
+#  -Dcom.sun.management.jmxremote.authenticate=false
+#  -Dcom.sun.management.jmxremote.port=29999
+#  -Dcom.sun.management.jmxremote.ssl=false
 # Master specific parameters. Default to TACHYON_JAVA_OPTS.
-export TACHYON_MASTER_JAVA_OPTS="$TACHYON_JAVA_OPTS"
+export TACHYON_MASTER_JAVA_OPTS="
+  $TACHYON_JAVA_OPTS
+  -Dcom.sun.management.jmxremote
+  -Dcom.sun.management.jmxremote.authenticate=false
+  -Dcom.sun.management.jmxremote.port=19997
+  -Dcom.sun.management.jmxremote.ssl=false
+  -Xloggc:$TACHYON_HOME/gclog.master.log
+"
 
 # Worker specific parameters that will be shared to all workers. Default to TACHYON_JAVA_OPTS.
-export TACHYON_WORKER_JAVA_OPTS="$TACHYON_JAVA_OPTS"
+export TACHYON_WORKER_JAVA_OPTS="$TACHYON_JAVA_OPTS
+  -Dcom.sun.management.jmxremote
+  -Dcom.sun.management.jmxremote.authenticate=false
+  -Dcom.sun.management.jmxremote.port=29997
+  -Dcom.sun.management.jmxremote.ssl=false
+  -Xloggc:$TACHYON_HOME/gclog.slave.log
+"
